app/models/article.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: articles
+#
+#  id           :bigint           not null, primary key
+#  for_banks    :boolean
+#  for_partners :boolean
+#  question     :string
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#
+class Article < ApplicationRecord
+  has_rich_text :content
+  validates :content, presence: true
+  validates :question, presence: true
+  validates :for_banks, acceptance: { message: "and for partners can't both be unchecked" }, unless: :for_partners
+  validates :for_partners, acceptance: { message: "and for banks can't both be unchecked" }, unless: :for_banks
+
+  def self.articles_for_banks(articles)
+    articles.select { |article| article&.for_banks }
+  end
```

Comment:
These methods should just be scopes:
```ruby
scope :for_banks, -> { where(for_banks: true) }
```

You can then add this scope to queries in other parts of the code.

Reply:
I'm unfamiliar with this, but I'll look into it. Thanks for the suggestion.